testonly/hammer/hammer.go

```diff
@@ -241,6 +288,26 @@ func newHammerState(ctx context.Context, cfg *MapConfig) (*hammerState, error) {
 	}, nil
 }
 
+// readChecker loops performing (read-only) checking operations until the done
+// channel is closed.
+func (s *hammerState) readChecker(ctx context.Context, done <-chan struct{}) error {
+	// Use a separate rand.Source so the main goroutine stays predictable.
+	prng := rand.NewSource(1)
```

Comment:
This will cause each reach checker to query the same items. Is that desired? 
Consider initializing with the task number: `prng := rand.NewSource(i)`

Reply:
Yeah, that's a better idea -- I was assuming the checkers would drift apart over time, as their timing got out of sync with the main goroutine.